### PR TITLE
[lite/micro] Override operator delete in memory planner

### LINKED ISSUE #32416: [lite/micro] missing delete() in GreedyMemoryPlanner
https://github.com/tensorflow/tensorflow/blob/8c0df1fa0b0490d8b1e54d7b019e2b2242ad6718/tensorflow/lite/experimental/micro/memory_planner/greedy_memory_planner.h#L43

does not override `void operator delete(void *p)` which results in link time error.

### PR DESCRIPTION
**Pull request by csukuangfj**
Fix #32416 .
Override operator delete.

### CODE REVIEW

**Review by wangtz — DISMISSED** [review]
Thanks! Do you also need this for linear_memory_planner.h?

**Comment by csukuangfj** [comment]
@wangtz 
`LinearMemoryPlanner` is never used so it does not cause link time error.
I'll fix it in the next patch.

**Review by wangtz — DISMISSED** [review]

**Comment by gbaned** [comment]
@csukuangfj Can you please resolve conflicts? Thanks!

**Comment by csukuangfj** [comment]
@gbaned 
Done.

**Comment by csukuangfj** [comment]
It takes nearly two months to merge such a small fix !!!

It's still not merged!

**Comment by wangtz** [comment]
@gbaned Hi Dero, could you help merge the PR?
Thanks!

**Review by wangtz — APPROVED** [review]

**Comment by gbaned** [comment]
@csukuangfj Can you please check build failures? Thanks!

**Comment by csukuangfj** [comment]
@gbaned 
The error log from <https://source.cloud.google.com/results/invocations/117384e4-7ee8-41a2-a56b-c09ba03c8320/targets/%2F%2Ftensorflow%2Ftools%2Fci_build%2Fbuilds:gen_android_out/log>
has nothing to do with my commit.
```
[0 / 80] [Prepa] BazelWorkspaceStatusAction stable-status.txt
ERROR: /tmpfs/tmp/bazel/external/jsoncpp_git/BUILD.bazel:5:1: C++ compilation of rule '@jsoncpp_git//:jsoncpp' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 29 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
gcc: error: unrecognized command line option '-std=c++14'
Target //tensorflow/core/common_runtime/eager:execute failed to build
Use --verbose_failures to see the command lines of failed build steps.
ERROR: /tmpfs/src/github/tensorflow/tensorflow/core/BUILD:242:1 C++ compilation of rule '@snappy//:snappy' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 27 argument(s) skipped)

```

**Comment by gbaned** [comment]
@csukuangfj Could you please address Ubuntu Sanity errors? Thanks!

**Comment by csukuangfj** [comment]
> ERROR: error loading package 'tensorflow/compiler/tests': Unable to load file '//tensorflow:tensorflow.google.bzl': file doesn't exist

from https://source.cloud.google.com/results/invocations/6070c286-565c-49ca-bdf3-df7db18f76e2/targets/%2F%2Ftensorflow%2Ftools%2Fci_build:gen_ci_sanity_out;shard=1;run=1;attempt=3/log

is not caused by this commit.

@gbaned 

I only added 4 lines and have tested it on my computer. Which line can make the CI failed ?

**Comment by gbaned** [comment]
@wangtz Could you PTAL. Thanks!

**Review by wangtz — DISMISSED** [review]
I don't see any chances on the code. Not sure why I need to Approve again, but here it is.

**Review by petewarden — DISMISSED** [review]

File: tensorflow/lite/experimental/micro/memory_planner/greedy_memory_planner.h [hunk 1]
```diff
@@ -16,6 +16,7 @@ limitations under the License.
 #ifndef TENSORFLOW_LITE_EXPERIMENTAL_MICRO_MEMORY_PLANNER_GREEDY_MEMORY_PLANNER_H_
 #define TENSORFLOW_LITE_EXPERIMENTAL_MICRO_MEMORY_PLANNER_GREEDY_MEMORY_PLANNER_H_
 
+#include "tensorflow/lite/experimental/micro/compatibility.h"
```
Comment (gbaned) [inline tensorflow/lite/experimental/micro/memory_planner/greedy_memory_planner.h]: > Here is the internal error we are getting , can you please check this.
/third_party/tensorflow/lite/experimental/micro/memory_planner/greedy_memory_planner.h:19:10: fatal error: 'third_party/tensorflow/lite/experimental/micro/compatibility.h' file not found
#include "third_party/tensorflow/lite/experimental/micro/compatibility.h"
cc @petewarden @wangtz

**Comment by mihaimaruseac** [comment]
Nit: Please don't include issue number in title, include it in the description of the PR. And give the PR a descriptive title

**Comment by gbaned** [comment]
@csukuangfj  Could you please address above internal error and resolve conflicts? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by csukuangfj** [comment]
@gbaned 
you've mentioned your two colleagues but neither of them replies.  I do NOT know
why the file `compatibility.h` is missing and I guess there must be some reasons for
its removal.

**Three** months have passed and such a small change is still not merged. I can think of the
high communication cost in such a large project.

I will resolve the conflicts tomorrow.

**Review by petewarden — APPROVED** [review]

**Comment by petewarden** [comment]
Thanks for your patience, sorry this has taken so long, we're trying to improve the process so it doesn't take so long in the future. Feel free to email me directly at petewarden@google.com if you find yourself stuck with a long-running PR like this again.

**Comment by rthadur** [comment]
Changes have been submitted internally , waiting for auto-merge to happen. Thank you

**Pull request merged**